### PR TITLE
feat: support rollup 1.x

### PR DESCRIPTION
**Pull request by skeggse**
Modernize the module's tooling: eslint, prettier, babel, rollup, jest. Rename `entry` to `input`, `useEntry` to `useInput`. Slightly improve caching behavior to avoid concurrent resolution of the same module.

BREAKING CHANGE: Supports Rollup 1.x, and likely does not support earlier versions. Change `entry` parameter to `input`, and `useEntry` to `useInput`. Uses newer untranspiled language features such as `String#includes`, but should otherwise retain parity with the previous version.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -10,12 +13,12 @@ var path = require('path');
 const isString = (value) => typeof value === 'string';
 
 /**
- * Coerce the input to an array of strings. If the input is a string, wrap it in
- * an array. If the input is a string, filter out all the non-string elements.
- * If we end up with an empty array, return null.
+ * Coerce the input to an array of strings. If the input is a string, wrap it in an array. If the
+ * input is a string, filter out all the non-string elements. If we end up with an empty array,
```
Comment (shils) [inline lib/index.js]:
```suggestion
 * input is an array, filter out all the non-string elements. If we end up with an empty array,
```

File: lib/index.js [hunk 2]
```diff
@@ -26,55 +29,43 @@ function strArray(array) {
 }
 
 /**
- * Call the evaluate function asynchronously on each item in the items array,
- * and return a promise that will fail on the first error evaluate produces, or
- * resolve to the first truthy value which evaluate produces. If all items
- * evaluate to falsy values, and evaluate never produces an error, the promise
- * will resolve to null.
+ * Call the evaluate function asynchronously on each item in the items array, and return a promise
+ * that will fail on the first error evaluate produces, or resolve to the first truthy value which
```
Comment (shils) [inline lib/index.js]:
```suggestion
 * that will reject with the first error evaluate produces, or resolve to the first truthy value which
```

File: lib/index.js [hunk 3]
```diff
@@ -83,73 +74,76 @@ class Import {
...
+        // If we've cached this import, don't bother trawling the filesystem again.
         const cached = cache[importee];
         if (cached) return cached;
 
-        return resolve(importee, imports).then((resolved) => {
-          // Save this in the cache in case we see it again.
-          cache[importee] = resolved;
-          return resolved;
-        });
+        const resolution = resolve(importee, imports);
```
Comment (shils) [inline lib/index.js:139]: WRT caching, do we have to consider the case where `resolution` rejects?

Comment (skeggse) [reply lib/index.js:139]: That's a good call, but I think we don't - if it rejects, then everything will come to a halt, and we won't have a poisoned cache.

File: lib/index.test.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  }).then(async (bundle) => {
+    const { output } = await bundle.generate({
+      format: 'cjs',
+    });
+    expect(output).toHaveLength(1);
+    return output[0].code;
+  });
+}
+
+function runModule(code) {
```
Comment (shils) [inline lib/index.test.js:30]: This seems like it should be in a standard library for rollup plugin testing

File: lib/index.test.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    ).resolves.toBe('no-extra'));
+
+  it('should try extensions in order', () =>
+    expect(
+      run('extension/main.js', {
+        extensions: ['.js', '.js.js'],
+      })
+    ).resolves.toBe('no-extra'));
+
+  it('should try extensions in order', () =>
```
Comment (shils) [inline lib/index.test.js:168]: Was this supposed to have the same title as the case above? It makes sense for both, just wondering

**Review by shils — APPROVED** [review]
It would be nice if we could test this on some open source projects and compare the outputs with that of the current version.

File: lib/index.test.js (hunk 4 above)
Comment (skeggse) [reply lib/index.test.js:30]: Heh yeah

File: lib/index.test.js (hunk 5 above)
Comment (skeggse) [reply lib/index.test.js:168]: :man_shrugging: these were the existing test names - I don't see much of a reason to make them different, though

**Pull request merged**